rllib/policy/policy.py

```diff
@@ -136,9 +145,9 @@ def serialize(self) -> Dict:
             "policy_class": cls,
             "observation_space": space_to_dict(self.observation_space),
             "action_space": space_to_dict(self.action_space),
-            # TODO(jungong) : try making the config dict durable by maybe
-            # getting rid of all the fields that are not JSON serializable.
-            "config": self.config,
+            # Make the config dict durable by getting rid of all the fields
+            # that are code (not JSON serializable).
+            "config": AlgorithmConfig._serialize_dict(self.config),
```

Comment:
will this fail if the dict is not json serializable? If so, where will it fail? When we try to msgpack.dump?

Reply:
no wait but my question is at what point will an error be raised if a config has non serializable data?

is there a failure? do we fail fast? does failing happen upon deserialization (that would be bad)

Reply:
The config only has two fields that are non-serializable: `policy_mapping_fn` and `is_policy_to_train`.
These are both callables that are very hard to serialize as they are usually provided via a lambda (as opposed to a proper class, which could be encoded as a classpath string). These two fields we always set to `__non_serializable__` and then require the user to provide those - again - once she tries to reinstate the algorithm from the respective checkpoint.

In my opinion, the `serialize()` should just not fail, b/c there are no other fields that could pose surprises here.
Does this make sense?

Reply:
ok so long as there aren't going to be any other non serializable fields this is should be predictable / reliable.